staging/src/k8s.io/client-go/tools/cache/delta_fifo.go

```diff
@@ -577,10 +577,11 @@ const (
 	Added   DeltaType = "Added"
 	Updated DeltaType = "Updated"
 	Deleted DeltaType = "Deleted"
-	// The other types are obvious. You'll get Sync deltas when:
-	//  * A watch expires/errors out and a new list/watch cycle is started.
-	//  * You've turned on periodic syncs.
-	// (Anything that trigger's DeltaFIFO's Replace() method.)
+	// Replaced is emitted when we encountered watch errors and had to do a
+	// relist. We don't here know if the newly relisted object has changed;
+	// shared informers will compare against their store to find out.
+	Replaced DeltaType = "Replaced"
```

Comment:
Old client may have written code that will not expect to see a new DeltaType.

Reply:
Ah. Yikes.